aws/data_source_aws_route53_delegation_set_test.go

```diff
@@ -0,0 +1,42 @@
+package aws
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceRoute53DelegationSet_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDataSourceAWSRoute53DelegationSetConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
```

Comment:
When checking data source attributes that should equal their resource counterparts, we prefer to use `resource.TestCheckResourceAttrPair()` to keep the tests easily maintainable, e.g.

```go
dataSourceName := "data.aws_route53_delegation_set.test"
resourceName := "aws_route53_delegation_set.test"
// ...
resource.TestCheckResourceAttrPair(dataSourceName, "caller_reference", resourceName, "caller_reference"),
resource.TestCheckResourceAttrPair(dataSourceName, "name_servers.#", resourceName, "name_servers.#"),
```

Reply:
Oops, this was also handled in my latest commit.